test/release/examples/primers/modules.chpl

```diff
@@ -0,0 +1,386 @@
+ *     unlimited
+ *     explicit exclusion of symbols
+ *     explicit inclusion of symbols
+ *     renaming included symbols
+ *   cross-file module access
+ */
+
+
+/* A module is a grouping of code - each program consists of at least one
+   module, and every symbol is associated with some module.  If all the code of
+   a file is not enclosed in an explicit module, defined using the 'module'
+   keyword, then the file itself is treated as a module with the same name as
+   the file (minus the .chpl suffix).  The compiler can be directed to include
+   modules in separate files by naming them on the command line, utilizing the
+   -M flag, etc. (see the man page for exact details).
+ */
+module modToUse {
+
+  /* In this case, foo is a public module-level variable that is defined within
+     the module modToUse */
+  var foo = 12;
+
+  /* As is bar. */
+  var bar: int = 2;
+
+  /* A symbol can be declared "private" - this means that only code defined
+     within the same scope as the definition of this symbol can access it.
+
+     Here, hiddenFoo is a private global variable, which is only accessible by
+     symbols defined within modToUse
+  */
+  private var hiddenFoo = false;
+
+
+  /* baz is a public function which is defined within modToUse */
+  proc baz (x, y) {
+    return x * (x + y);
+  }
+
+  /* hiddenBaz is a private function, which is also only accessible by symbols
+     defined within modToUse.
+  */
+  private proc hiddenBaz(a) {
+    writeln(a);
+    return a + 3;
+  }
+
+  // In the current implementation, private cannot be applied to type
+  // definitions; type aliases, and declarations of enums, records, and
+  // classes cannot be declared private.  Private also cannot be applied to
+  // fields or methods yet.
+
+}
+
+module AnotherModule {
+  var a = false;
+}
+
+module ThirdModule {
+  var b = -13.0;
+}
+
+
+module Conflict {
+  /* This variable shares a name with a symbol in modToUse. */
+  var bar = 5;
+
+  var other = 5.0 + 3i;
+
+  var another = false;
+}
+
+
+module DifferentArguments {
+  /* This function shares a name with a function in modToUse, but requires
+     different arguments */
+  proc baz(x) {
+    return x - 2;
+  }
+}
+
+module MainModule {
+  proc main() {
+    writeln("Access from outside a module");
+
+    /* If a module is not the main module for a program, it is desirable for its
+       contents to be accessible to external modules.  There are several
+       strategies for accomplishing this:
+
+       First, a symbol can be referenced explicitly - this is done using the
+       module name and a separating '.' as a prefix to the name of the symbol
+       desired.
+    */
+    var thriceFoo = 3 * modToUse.foo; // should be 36
+    writeln(thriceFoo);
+
+
+    {
+      /* If many of the module's symbols are desired, or the same symbol is
+         desired multiple times, then it is best to utilize what is known as a
+         "use statement".
```

Comment:
"many" -> "several"
"it is best" -> "you may want to use" or "it can be convenient to use"?  (a Kyle-style mentality might say that fully qualified is always best, period).